pkg/operator/storageclasscontroller/storageclasscontroller.go

```diff
@@ -125,14 +133,15 @@ func (c *StorageClassController) Sync(ctx context.Context, connection *vclib.VSp
 func (c *StorageClassController) syncStoragePolicy(ctx context.Context, connection *vclib.VSphereConnection, apiDeps checks.KubeAPIInterface, scState operatorapi.StorageClassStateName) (string, checks.ClusterCheckResult) {
 	// if the SC is not managed, there is no need to sync the storage policy
 	if !c.scStateEvaluator.IsManaged(scState) {
+		klog.V(2).Info("sc is not managed")
 		return "", checks.MakeClusterCheckResultPass()
 	}
 
 	// if we are running the checks after creating the policy successfully
 	// then lets run checks less frequently.
-	if !time.Now().After(c.nextCheck) && len(c.policyName) > 0 {
+	if !time.Now().After(c.nextCheck) && len(c.connPolicyNames[connection.Hostname]) > 0 {
 		klog.V(4).Infof("Returning without running any checks")
-		return c.policyName, checks.MakeClusterCheckResultPass()
+		return c.connPolicyNames[connection.Hostname], checks.MakeClusterCheckResultPass()
```

Comment:
We will have to now probably create multiple storageclasses. One for each vcenter.

Reply:
Right.  The connPolicyNames is containing policy for each vCenter that we create.  Originally, first vCenter to create the policy set this older variable which made logic think second vCenter was already done.  So with this map, we are storing fact that each vCenter policy creation is complete.